elements/lisk-p2p/test/unit/peer/inbound.ts

```diff
@@ -13,67 +13,229 @@
  *
  */
 import { expect } from 'chai';
-import { SCServerSocket } from 'socketcluster-server';
-import { InboundPeer } from '../../../src/peer';
+import { InboundPeer, PeerConfig } from '../../../src/peer';
 import { P2PDiscoveredPeerInfo } from '../../../src/p2p_types';
+import { SCServerSocket } from 'socketcluster-server';
+import {
+	DEFAULT_RANDOM_SECRET,
+	DEFAULT_PING_INTERVAL_MAX,
+	DEFAULT_PING_INTERVAL_MIN,
+} from '../../../src/constants';
+import {
+	REMOTE_SC_EVENT_MESSAGE,
+	REMOTE_SC_EVENT_RPC_REQUEST,
+	REMOTE_EVENT_PING,
+} from '../../../src/events';
 
 describe('peer/inbound', () => {
-	const DEFAULT_RANDOM_SECRET = 123;
-	const defaultPeerInfo: P2PDiscoveredPeerInfo = {
-		ipAddress: '12.12.12.12',
-		wsPort: 5001,
-		height: 545776,
-		isDiscoveredPeer: true,
-		version: '1.1.1',
-		protocolVersion: '1.1',
-	};
-
-	const socket = <SCServerSocket>({
-		on: sandbox.stub(),
-	} as any);
-
-	const defaultPeer = new InboundPeer(defaultPeerInfo, socket, {
-		rateCalculationInterval: 1000,
-		wsMaxMessageRate: 1000,
-		wsMaxMessageRatePenalty: 10,
-		secret: DEFAULT_RANDOM_SECRET,
-		maxPeerInfoSize: 10000,
-		maxPeerDiscoveryResponseLength: 1000,
+	let defaultPeerInfo: P2PDiscoveredPeerInfo;
+	let defaultPeerConfig: PeerConfig;
+	let defaultInboundPeer: InboundPeer;
+	let inboundSocket: SCServerSocket;
+	let clock: sinon.SinonFakeTimers;
+
+	beforeEach(() => {
+		clock = sandbox.useFakeTimers();
+		defaultPeerInfo = {
+			ipAddress: '12.12.12.12',
+			wsPort: 5001,
+			height: 545776,
+			isDiscoveredPeer: true,
+			version: '1.1.1',
+			protocolVersion: '1.1',
+		};
+		defaultPeerConfig = {
+			rateCalculationInterval: 1000,
+			wsMaxMessageRate: 1000,
+			wsMaxMessageRatePenalty: 10,
+			secret: DEFAULT_RANDOM_SECRET,
+			maxPeerInfoSize: 10000,
+			maxPeerDiscoveryResponseLength: 1000,
+		};
+		inboundSocket = <SCServerSocket>({
+			on: sandbox.stub(),
+			off: sandbox.stub(),
+			emit: sandbox.stub(),
+			destroy: sandbox.stub(),
+		} as any);
+		defaultInboundPeer = new InboundPeer(
+			defaultPeerInfo,
+			inboundSocket,
+			defaultPeerConfig,
+		);
+	});
+
+	afterEach(() => {
+		clock.restore();
+		defaultInboundPeer.disconnect();
 	});
 
 	describe('#constructor', () => {
-		it('should be an object', () => {
-			return expect(defaultPeer).to.be.an('object');
+		it('should be an instance of InboundPeer class', () =>
```

Comment:
for this and all other `it` blocks before, please  `return expect` or make callback `async`

Reply:
Why shall we do this? If there is any strong reasoning, we should add a Eslint rule. @shuse2 @jondubois could you share your thoughts on this?

Reply:
I thought we already have one, but if something is async, the test will be false positive.
Therefore, we should consistently use `async`. (this rule was there for a long time in elements, so we should add the rule back if it doesn't work

Reply:
Let's open a new issue to investigate and apply that across every test.

Reply:
I don't have a strong opinion. I think in other parts of the project were were following this principle because otherwise the linter would complain. If the linter doesn't complain I think it's OK but I guess @shuse2 would have a better idea.